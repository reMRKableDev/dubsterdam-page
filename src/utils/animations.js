import { keyframes } from "styled-components";
import { bounce, fadeInUp, fadeInDown, shake, merge } from "react-animations";

export const bounceAnimation = keyframes`${bounce}`;
export const fadeInUpAnimation = keyframes`${fadeInUp}`;
export const fadeInDownAnimation = keyframes`${fadeInDown}`;
export const shakeAnimation = keyframes`${shake}`;
export const fadeInUpShakeAnimation = keyframes`${merge(fadeInUp, shake)}`;
