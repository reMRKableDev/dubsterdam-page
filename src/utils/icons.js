import { Mail, Github, Linkedin, Database } from "@styled-icons/feather";
import {
  Css3,
  Html5,
  ReactLogo,
  Nodejs,
  Javascript,
} from "@styled-icons/boxicons-logos";

export const mailIcon = Mail;
export const gitHubIcon = Github;
export const linkedInIcon = Linkedin;
export const databaseIcon = Database;
export const cssIcon = Css3;
export const htmlIcon = Html5;
export const reactIcon = ReactLogo;
export const nodeIcon = Nodejs;
export const javascriptIcon = Javascript;
