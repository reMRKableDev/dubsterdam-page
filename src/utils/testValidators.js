export const validateNotEmpty = (received) => {
  expect(received).not.toBeNull();
  expect(received).toBeTruthy();
};

export const validateDarkThemeMode = (received) => {
  expect(received).not.toMatch(/light/i);
  expect(received).toMatch(/dark/i);
};

export const validateLightThemeMode = (received) => {
  expect(received).not.toMatch(/dark/i);
  expect(received).toMatch(/light/i);
};
