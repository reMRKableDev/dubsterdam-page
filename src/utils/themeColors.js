import theme from "styled-theming";

export const bgColor = theme("mode", {
  light: "#fafafa",
  dark: "#222",
});

export const footerColor = theme("mode", {
  light: "#222",
  dark: "#fafafa",
});

export const textColor = theme("mode", {
  light: "#000",
  dark: "#fff",
});

export const buttonBgColor = theme("mode", {
  light: "#222",
  dark: "#eee",
});

export const buttonTextColor = theme("mode", {
  light: "#eee",
  dark: "#222",
});

export const brandColor = theme("mode", {
  light: "#ed6c6d",
  dark: "#ed6c6d",
});

export const textAccentColor = theme("mode", {
  light: "#555",
  dark: "#fff",
});
