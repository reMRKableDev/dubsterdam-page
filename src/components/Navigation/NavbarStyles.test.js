import React from "react";
import { shallow } from "enzyme";
import {
  StyledNav,
  StyledThemeButton,
  StyledThemeButtonContainer,
} from "./NavbarStyles";

import { validateMatchSnapshot } from "../../utils/testValidators";

describe("Navbar Styles unit tests", () => {
  it("should validate match to snapshot of StyledNav", () => {
    const wrapper = shallow(<StyledNav />);
    validateMatchSnapshot(wrapper);
  });

  it("should validate match to snapshot of StyledThemeButton ", () => {
    const wrapper = shallow(<StyledThemeButton />);
    validateMatchSnapshot(wrapper);
  });

  it("should validate match to snapshot of StyledThemeButtonContainer ", () => {
    const wrapper = shallow(<StyledThemeButtonContainer />);
    validateMatchSnapshot(wrapper);
  });
});
