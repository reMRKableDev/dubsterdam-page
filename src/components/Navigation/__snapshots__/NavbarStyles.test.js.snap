// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar Styles unit tests should validate match to snapshot of StyledNav 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  position: fixed;
  z-index: 3;
  width: 100%;
}

<nav
  className="c0"
/>
`;

exports[`Navbar Styles unit tests should validate match to snapshot of StyledThemeButton  1`] = `
.c0 {
  border: 1px solid;
  box-shadow: 0 2px 8px rgba(0,0,0,0.3),0 0 40px rgba(0,0,0,0.1) inset;
  cursor: pointer;
  font-size: 0.8em;
  padding: 0.5em 1em;
  outline: none;
}

<button
  className="c0"
/>
`;

exports[`Navbar Styles unit tests should validate match to snapshot of StyledThemeButtonContainer  1`] = `
.c0 {
  padding: 10px;
}

<div
  className="c0"
/>
`;
